### ISSUE TITLE
emqx docker 4.2.6 webhook shutdown,{options,{keyfile,[]}}

### ISSUE THREAD

**Issue by rauntska** [emqx/emqx#4029]
I seem to be having some error with the latest 4.2.6 emqx version.
Image im running : `image: emqx/emqx:4.2.6`

emqx_web_hook.conf:

```
##====================================================================
## WebHook
##====================================================================

## The web services URL for Hook request
##
## Value: String
web.hook.api.url = https://blahblah.azurewebsites.net/api/MqttEventLogger

##--------------------------------------------------------------------
## HTTP Request Headers
##
## The header params what you extra need
## Format:
##    web.hook.headers.<param> = your-param
## Example:
## 1. web.hook.headers.token = your-token
## 2. web.hook.headers.other = others-param
##
## Value: String
## web.hook.headers.token = your-token

##--------------------------------------------------------------------
## Encode message payload field
##
## Value: base64 | base62
## web.hook.encode_payload = base64
## Mysql ssl configuration.
##
## Value: on | off
## web.hook.ssl = off

##--------------------------------------------------------------------
## CA certificate.
##
## Value: File
## web.hook.ssl.cafile  = path to your ca file
## Client ssl certificate.
##
## Value: File
## web.hook.ssl.certfile = path to your clientcert file

##--------------------------------------------------------------------
## Client ssl keyfile.
##
## Value: File
## web.hook.ssl.keyfile = path to your clientkey file

##--------------------------------------------------------------------
## Pool size of HTTP connection pools.
##
## Value: Integer
# web.hook.pool_size = 32

##--------------------------------------------------------------------
## Hook Rules
## These configuration items represent a list of events should be forwarded
##
## Format:
##   web.hook.rule.<HookName>.<No> = <Spec>
web.hook.rule.client.connect.1       = {"action": "on_client_connect"}
web.hook.rule.client.connack.1       = {"action": "on_client_connack"}
web.hook.rule.client.connected.1     = {"action": "on_client_connected"}
web.hook.rule.client.disconnected.1  = {"action": "on_client_disconnected"}
web.hook.rule.client.subscribe.1     = {"action": "on_client_subscribe"}
web.hook.rule.client.unsubscribe.1   = {"action": "on_client_unsubscribe"}
web.hook.rule.session.subscribed.1   = {"action": "on_session_subscribed"}
web.hook.rule.session.unsubscribed.1 = {"action": "on_session_unsubscribed"}
web.hook.rule.session.terminated.1   = {"action": "on_session_terminated"}
web.hook.rule.message.publish.1      = {"action": "on_message_publish"}
web.hook.rule.message.delivered.1    = {"action": "on_message_delivered"}
web.hook.rule.message.acked.1        = {"action": "on_message_acked"}
```
Seems like when it tries to trigger the webhook it keeps getting:
```
`  2021-01-18 06:54:45.184 [error] <<"ServiceConsumer">> [WebHook] HTTP request error: {shutdown,{options,{keyfile,[]}}}

 2021-01-18 06:54:50.185 [error] <<"ServiceConsumer">> [WebHook] HTTP request error: timeout

  2021-01-18 06:54:50.189 [error] <<"ServiceConsumer">> [WebHook] HTTP request error: {shutdown,{options,{keyfile,[]}}}

   2021-01-18 06:54:55.190 [error] <<"ServiceConsumer">> [WebHook] HTTP request error: timeout

    2021-01-18 06:54:55.195 [error] <<"ServiceConsumer">> [WebHook] HTTP request error: {shutdown,{options,{keyfile,[]}}}`
```

The url itself works fine (tested with postman)

BR,
Rauno

**Comment by tigercl** [comment]
@rauntska Thank you for your feedback, we will fix this issue as soon as possible.

**Comment by HJianBo** [comment]
In version 4.2.6, two workarounds are
1. do not use https
2. use two-way authentication and properly configure the three options on cafile, certfile, keyfile.

**Comment by HJianBo** [comment]
Hi @rauntska Thank you very much for your feedback. We have fixed this issue in 4.2.7 and you are welcome to verify it again.
